galsim/chromatic.py

```diff
@@ -2027,6 +2027,47 @@ def evaluateAtWavelength(self, wave):
         return galsim.AutoCorrelate(self.obj.evaluateAtWavelength(wave), **self.kwargs)
 
 
+class ChromaticFourierSqrtProfile(ChromaticObject):
+    """A class for computing the Fourier-space square root of a ChromaticObject.
+
+    The ChromaticFourierSqrt class represents a wavelength-dependent Fourier-space square root of a profile.
+
+    You may also specify a gsparams argument.  See the docstring for GSParams using
+    help(galsim.GSParams) for more information about this option.  Note: if `gsparams` is
+    unspecified (or None), then the ChromaticFourierSqrtProfile instance inherits the same GSParams as
+    the object being operated on.
+
+    Initialization
+    --------------
+
+    @param obj              The object to compute the Fourier-space square root of.
+    @param gsparams         An optional GSParams argument.  See the docstring for GSParams for
+                            details. [default: None]
+    """
+    def __init__(self, obj, **kwargs):
+        self.obj = obj.copy()
```

Comment:
I think you probably just copied this from `ChromaticDeconvolution`, but I'm wondering if the `.copy()` is actually necessary since all `ChromaticObject`s should be immutable?  If it's not needed, no need to change here, but I wanted to flag this for a future "cleanup the copying" issue.